### PR TITLE
Remove duplicates from the DARC get build output

### PR DESCRIPTION
**Pull request by ali-turan7**
https://github.com/dotnet/arcade/issues/12933

### CODE REVIEW

**Comment by dkurepa** [comment]
Could you please write why we're getting duplicates?

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs [hunk 1]
```diff
@@ -57,14 +57,16 @@ public override async Task<int> ExecuteAsync()
...
+                    .Where(r => r.Contains(_options.Repo, StringComparison.OrdinalIgnoreCase))                    
                     .ToHashSet(StringComparer.OrdinalIgnoreCase);
 
                 matchingBuilds = new List<Build>();
                 foreach (string repo in possibleRepos)
                 {
                     matchingBuilds.AddRange(await remote.GetBuildsAsync(repo, _options.Commit));
                 }
+
+                matchingBuilds = matchingBuilds.GroupBy(build => UxHelpers.GetTextBuildDescription(build), build => build).Select(group => group.First()).ToList();
```
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs:69]: Ahh yes, we should be able to just do `matchingBuilds = matchingBuilds.DistinctBy(build => UxHelpers.GetTextBuildDescription(build))` I forgot about that

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+        GetBuildOperation getBuildOperation = new(options, _services);
+
+        int result = await getBuildOperation.ExecuteAsync();
+
+        result.Should().Be(Constants.SuccessCode);
+
+        var output = _consoleOutput.GetOuput();
+        output.Should().Be(UxHelpers.GetTextBuildDescription(build));
+
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:148]: nit: remove empty line

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            assets: ImmutableList.Create<Asset>(),
+            dependencies: ImmutableList.Create<BuildRef>(),
+            incoherencies: ImmutableList.Create<BuildIncoherence>()
+            )
+        {
+            AzureDevOpsRepository = internalRepo,
+            GitHubRepository = githubRepo,
+        };
+
+
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:129]: nit: remove extra line

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        GetBuildOperation getBuildOperation = new(options, _services);
+
+        int result = await getBuildOperation.ExecuteAsync();
+
+        result.Should().Be(Constants.SuccessCode);
+
+        var output = _consoleOutput.GetOuput();
+        output.Should().Be(UxHelpers.GetTextBuildDescription(build));
+
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:100]: nit: remove extra line

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs (hunk 2 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:148]:
```suggestion
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs (hunk 3 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:129]:
```suggestion
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs (hunk 4 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:100]:
```suggestion
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs [hunk 5]
```diff
@@ -0,0 +1,173 @@
+using FluentAssertions;
```
[the remark this answers is not in the document]
Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:1]: Ahh yes, I forgot. Ali, you can just copy the header from any other file in this repo

[the remark this answers is not in the document]
Comment (ali-turan7) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:7]: Done.

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Darc.Tests.Operations;
```
[the remark this answers is not in the document]
Comment (ali-turan7) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/Operations/GetBuildOperationTests.cs:22]: Done

**Review by dkurepa — APPROVED** [review]

**Pull request merged**